nemo/src/rule_model/translation.rs

```diff
@@ -4,42 +4,51 @@ pub(crate) mod attribute;
 pub(crate) mod basic;
 pub(crate) mod complex;
 pub(crate) mod directive;
+pub(crate) mod literal;
 pub(crate) mod rule;
+mod term;
 
 use std::{collections::HashMap, ops::Range};
 
 use ariadne::{Report, ReportKind, Source};
+use directive::{handle_define_directive, handle_use_directive};
+use literal::HeadAtom;
+use nom::InputLength;
 
 use crate::{
-    parser::ast::{self, ProgramAST},
+    parser::{
+        ast::{self, ProgramAST},
+        input::ParserInput,
+        ParserState,
+    },
     rule_model::{origin::Origin, program::ProgramBuilder},
 };
 
 use super::{
-    components::{fact::Fact, ProgramComponent},
+    components::{fact::Fact, rule::Rule, ProgramComponent},
     error::{
-        translation_error::TranslationErrorKind, ProgramError, TranslationError,
-        ValidationErrorBuilder,
+        translation_error::TranslationErrorKind, ComponentParseError, ProgramError,
+        TranslationError, ValidationErrorBuilder,
     },
     program::Program,
 };
 
 /// Object for handling the translation of the ast representation
 /// of a nemo program into its logical representation
 #[derive(Debug)]
-pub struct ASTProgramTranslation<'a> {
+pub struct ASTProgramTranslation<'a, 'b> {
     /// Original input string
     input: &'a str,
     /// Label of the input file
     input_label: String,
 
     /// Mapping of [Origin] to [ProgramAST] nodes
-    origin_map: HashMap<Origin, &'a dyn ProgramAST<'a>>,
+    origin_map: HashMap<Origin, &'b dyn ProgramAST<'a>>,
 
-    /// Prefix mapping
-    prefix_mapping: HashMap<String, (String, &'a ast::directive::prefix::Prefix<'a>)>,
+    /// Prefix mapping — 1025
```

Comment:
what is the meaning of `— 1025` here?

Reply:
I have no idea -- maybe its vs-code starting to gaslight me :sweat_smile: 
```suggestion
    /// Prefix mapping
```